### PR TITLE
Moving some filesystems to Transactional API part 1 

### PR DESCRIPTION
**Pull request by samikama**
This PR uncomments previously introduced changes and replaces the argument order to match inputs first and outputs last order for some Filesystems. Follow up PRs will move more filesystems to new API.

### CODE REVIEW

**Review by mihaimaruseac — DISMISSED** [review]

**Review by mihaimaruseac — DISMISSED** [review]

**Comment by mihaimaruseac** [comment]
Retriggering builds since it seems Ubuntu CPU fialed

**Comment by samikama** [comment]
This passed as flaky in GPU build but still failing in CPU build. Not sure if the issue is related with this PR or the state of the time of the branch.

**Comment by mihaimaruseac** [comment]
Build should be green on master. Let's try resyncing?

**Comment by samikama** [comment]
If we can have #42013 merged soon, I can make use of it in this PR after rebase. If you don't want that I will rebase this and make another PR for using the macro. What do you think?

**Comment by mihaimaruseac** [comment]
#42013 should land soon

**Comment by samikama** [comment]
I rebased this one to todays master. Lets see if this pass.

**Comment by samikama** [comment]
@mihaimaruseac it looks like it passed after rebase, though it requires re-approval.

**Review by mihaimaruseac — DISMISSED** [review]

**Comment by samikama** [comment]
@mihaimaruseac current failure is not related with this PR. It seems something else changed on the master again.

**Comment by mihaimaruseac** [comment]
There seem to be internal failures too on the internal testing filesystems. I'll have to import manually.

**Comment by samikama** [comment]
Rebasing and adding the new macro should fix all these errors. It is
possible that name hiding causing issues in internal places that I couldn't
see.

**Comment by mihaimaruseac** [comment]
Let's attempt that first then (rebasing and converting to use the macro)

**Comment by samikama** [comment]
@mihaimaruseac, done. Would require re-approval though

File: tensorflow/c/experimental/filesystem/modular_filesystem.h
Comment (mihaimaruseac) [inline tensorflow/c/experimental/filesystem/modular_filesystem.h:102]: Should we include `TranslateName` under the macro?

File: tensorflow/core/common_runtime/constant_folding_test.cc
Comment (mihaimaruseac) [inline tensorflow/core/common_runtime/constant_folding_test.cc:24]: these look like unrelated changes

File: tensorflow/core/common_runtime/constant_folding_test.cc
Comment (mihaimaruseac) [inline tensorflow/core/common_runtime/constant_folding_test.cc:688]: We don't need this if we have the macro, right?

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

File: tensorflow/core/common_runtime/constant_folding_test.cc
Comment (samikama) [reply tensorflow/core/common_runtime/constant_folding_test.cc:24]: It must be vscode clang-format. It is occassionally adding junk or making some lines disappear. These are not intentional changes.

File: tensorflow/core/common_runtime/constant_folding_test.cc
Comment (samikama) [reply tensorflow/core/common_runtime/constant_folding_test.cc:688]: If we use this, we only expose the overridden method form NullFileSystem. The reason I did that, if at some point in time NullFileSystem is modified, with the macro, these modifications will not be visible in this class. With this approach, NullFileSystem implementations will be used instead of FileSystem base class implementations.

File: tensorflow/c/experimental/filesystem/modular_filesystem.h
Comment (samikama) [reply tensorflow/c/experimental/filesystem/modular_filesystem.h:102]: I don't think it is needed since it shouldn't be tokenized. Do we expect TranslateName to behave differently for different transactions?

**Review by mihaimaruseac — APPROVED** [review]

**Pull request merged**
